### PR TITLE
Do not include Facebook license on users codegen'd code

### PR DESCRIPTION
**Pull request by acoates-ms**
## Summary

The codegen generates a Facebook copyright notice at the top of the generated files.  While this might make sense on the core files, this codegen will be run on external components too.   The notice also refers to a LICENSE file in the root of this project, which might not be there if this is run on another project.   I did a quick look at some of the codegen that we ship within windows dev tools, and it looks like we normally just have comments saying the file was codegen'd and so the file shouldn't be manually edited.

Open to suggestions on what the comment header should say.

## Changelog

[General] [Changed] - Do not include Facebook license on users codegen'd code

### CODE REVIEW

**Comment by hramos** [comment]
Thanks for the PR. Let me run this through our OSS IP team and I'll get back to you.

**Review by hramos — APPROVED** [review]

File: packages/react-native-codegen/src/generators/components/GenerateComponentDescriptorH.js [hunk 1]
```diff
@@ -17,10 +17,10 @@ type FilesOutput = Map<string, string>;
 
 const template = `
 /**
- * Copyright (c) Facebook, Inc. and its affiliates.
+ * This code was generated by a tool.
```
Comment (cortinico) [inline packages/react-native-codegen/src/generators/components/GenerateComponentDescriptorH.js:20]:
```suggestion
 * This code was generated by [react-native-codegen](https://www.npmjs.com/package/react-native-codegen).
```

File: packages/react-native-codegen/src/generators/components/GenerateComponentDescriptorH.js [hunk 2]
```diff
@@ -17,10 +17,10 @@ type FilesOutput = Map<string, string>;
...
  *
- * This source code is licensed under the MIT license found in the
- * LICENSE file in the root directory of this source tree.
+ * Changes to this file may cause incorrect behavior and will be lost if
+ * the code is regenerated.
```
Comment (cortinico) [inline packages/react-native-codegen/src/generators/components/GenerateComponentDescriptorH.js:23]:
```suggestion
 * Do not edit this file as changes may cause incorrect behavior and will be lost
 * once the code is regenerated.
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Hey @acoates-ms,
Thanks for the PR. I'd like to review it and drive it to landing if you wish. 

Can I ask you to:
- Rebase on top of `main`
- Apply the changes I suggested to all the templates?

**Comment by cortinico** [comment]
@acoates-ms I've published this https://github.com/facebook/react-native/pull/32840 that is going to close this PR as it has several merge conflicts + The files inside `src/generators/modules` were missing. So this updates the header for all the generated files 👍

**Pull request closed without merging**